### PR TITLE
[BugFix] Correct Masterful Blitz count evaluation in Riddle of Fire

### PR DESCRIPTION
**Pull request by AkairyuGestalter**
The BlitzEvaluator's calculation for how many Masterful Blitz actions were expected to be used in a Riddle of Fire window was not properly taking into account whether the player was already in a Perfect Balance window when Riddle of Fire was activated. Created a function in the Perfect Balance module to calculate that based on either having the status or having contents in the beast Chakra gauge. Also cleaned up the available charge calculations by using the cooldowns module's charge history instead of trying to replicate it.

Note that this builds on top of the changes added in PR #1844 so approving this one will also pull through the changes from that one. Approve/merge that one first if you want to see the minimal set of changes this PR adds.

### CODE REVIEW

File: src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx
Comment (Toastdeib) [inline src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx:136]: Nit - can you confirm whether this comment is accurate? It says 3 but the `MAX_EXPECTED_BLITZES` const is 2.

File: src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx
Comment (Toastdeib) [inline src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx]: I *think* this should work with `this.cooldowns.chargeHistory(this.data.actions.PERFECT_BALANCE)` if I'm reading the TS correctly in the Cooldowns module, which (at least in my book) would be cleaner and safer than using the string key. That said, I do see some precedent elsewhere in these files for using a string instead of the action, so you're welcome to take this with a grain of salt if it's a standard practice.

**Review by Toastdeib — COMMENTED** [review]
I won't pretend to understand all of the math, but overall this looks fine to my untrained eye. I left a couple of small comments but I'm happy to approve and merge this once they're addressed as long as no one else objects.

File: src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx
Comment (AkairyuGestalter) [reply src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx:136]: Three is possible in the case where a blitz is prepared ahead of time, but the logic to try to reflect that was bad. Added a specific const for that case and polished the returns to make it more readable and accurate.

File: src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx
Comment (AkairyuGestalter) [reply src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx]: Yeah, I like that better. Have to pass it through from the RoF analyser since evaluators don't have direct access to `data` but still.

File: src/parser/jobs/mnk/modules/PerfectBalance.tsx
Comment (Toastdeib) [inline src/parser/jobs/mnk/modules/PerfectBalance.tsx:151]: Oop, good catch - I didn't spot this one in my initial review.

File: src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx
Comment (Toastdeib) [inline src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx:117]: 👍

**Review by Toastdeib — APPROVED** [review]
I think this is good to go now; I'll hold off on merging until this evening just in case anyone else wants to weigh in but the fixes all seem solid (and I even put in the effort to understand the math a bit).

**Pull request merged**
